modules/ppcp-wc-gateway/src/Assets/FraudNetAssets.php

```diff
@@ -168,19 +170,64 @@ protected function are_buttons_enabled_for_context() : bool {
 		if ( ! in_array( PayPalGateway::ID, $this->enabled_ppcp_gateways, true ) ) {
 			return false;
 		}
+		try {
+			$button_locations = $this->settings->get( 'smart_button_locations' );
+		} catch ( NotFoundException $exception ) {
+			return false;
+		}
 
-		$location_prefix             = $this->context === 'checkout' ? '' : "{$this->context}_";
-		$setting_name                = "button_{$location_prefix}enabled";
-		$buttons_enabled_for_context = $this->settings->has( $setting_name ) && $this->settings->get( $setting_name );
+		if ( $this->context() === 'pay-now' ) {
+			return true;
+		}
 
-		if ( $this->context === 'product' ) {
-			return $buttons_enabled_for_context || $this->settings->has( 'mini-cart' ) && $this->settings->get( 'mini-cart' );
+		if ( $this->context() === 'product' ) {
+			return in_array( 'product', $button_locations, true ) || in_array( 'mini-cart', $button_locations, true );
 		}
 
-		if ( $this->context === 'pay-now' ) {
-			return true;
+		return in_array( $this->context(), $button_locations, true );
+	}
+
+	/**
+	 * Get context.
+	 *
+	 * @return string
+	 */
+	protected function context(): string {
```

Comment:
Would be great to extract this somewhere to avoid duplication (currently here and in `SmartButton`). Maybe into a trait.